app/services/allocations/update.rb

```diff
@@ -0,0 +1,36 @@
+module Allocations
+  class Update
+    attr_accessor :object, :params
+
+    def initialize(allocation, params)
+      @object = allocation
+      @params = params
+
+      set_number_of_places
+      set_request_type
+
+      object
+    end
+
+    def execute
+      object.save
+    end
+
+  private
+
+    def set_number_of_places
+      object.number_of_places = case params[:request_type]
+                                when "declined"
+                                  0
+                                when "repeat"
+                                  object.previous&.number_of_places || 0
+                                else
+                                  params[:number_of_places]
```

Comment:
so if its not

`declined` or `repeat` what is is it?

https://github.com/DFE-Digital/teacher-training-api/blob/3dbee3b7ef813cd7006927e9c7a75315d2b0d787/spec/factories/allocations.rb#L7-L12

Feels head scratching it is missing a trait as well as a test for the other `request_type`

Reply:
The third option for request_type is `initial` but we're going to be addressing this in another PR. Updating an initial request to set a different number of places is going to require some further logic here, and we aren't quite sure yet what it'll look like.